packages/itwinui-css/src/tree/tree.scss

```diff
@@ -2,19 +2,54 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @use '../mixins';
 
-$iui-active-outline: thin solid var(--iui-color-border-accent);
-
-// padding of small borderless button + 1px transparent border
-$iui-expander-inline-padding: calc(var(--iui-size-2xs) + 1px);
-
-// width of small borderless button (button is 1:1 ratio)
-$iui-expander-button-width: calc(var(--iui-component-height-small) + 2px);
+$iui-active-outline: 1px solid var(--iui-color-border-accent);
 
 .iui-tree {
   @include mixins.iui-reset;
+  --_iui-expander-button-width: calc(var(--iui-component-height-small) + 2px);
+  --_iui-tree-node-padding-inline: var(--iui-size-xs);
+  --_iui-tree-row-height: calc(var(--iui-size-3xs) * 18);
+
   list-style: none;
   border-radius: var(--iui-border-radius-1);
   background-color: var(--iui-color-background);
+
+  &[data-iui-size='small'] {
+    --_iui-tree-node-padding-inline: var(--iui-size-2xs);
+    --_iui-tree-row-height: calc(var(--iui-size-3xs) * 14);
+  }
+
+  &[data-iui-lines='true'] {
+    isolation: isolate;
+
+    &:hover,
+    &:focus-within {
+      --_iui-tree-line-color: var(--iui-color-border-subtle);
+    }
+
+    .iui-sub-tree {
+      position: relative;
+
+      &::before {
+        content: '';
+        inline-size: 1px;
+        block-size: 100%;
+        position: absolute;
+        inset: 0 auto 0
+          calc(
+            var(--_iui-expander-button-width) * 0.5 * var(--level, 0) +
+            (var(--iui-size-m) + var(--_iui-tree-node-padding-inline) * 2)
+          ); // TODO: The last part is added for checkboxes, how does this work without checkboxes? 🤔
+        background-color: var(--_iui-tree-line-color, transparent);
+        transition: background-color var(--iui-duration-1) ease;
+        z-index: 1; // TODO: This gets the lines above the background hover, but not above the buttons when focused
+      }
+    }
+
+    .iui-button {
+      z-index: 2;
```

Comment:
Really would like to avoid `z-index`.  The 2 instances of `z-index` address:
1. When hovering over a row, the hover background color was covering up the lines.
2. When focusing the expander button, the line was covering up the focus outline.

Reply:
use `z-index: -1` on the pseudo-element?

Reply:
The 2nd point doesn't seem to be an issue anymore, my guess is reverting back to the small button for all tree sizes makes this not an issue anymore.

I'll look into the 1st point tomorrow, here is a screenshot of the bug when all `z-index`s are removed.
![Screenshot 2023-12-05 at 5 17 49 PM](https://github.com/iTwin/iTwinUI/assets/849817/7a2c5c8c-fe13-4745-8653-9decc544d1fe)